### PR TITLE
Feature/refactor item authoring local configuration

### PR DESCRIPTION
**Pull request by bdrainer**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -218,9 +218,6 @@ services:
       ITEMBANK_SERVICES_HISTORY_URL: http://host.docker.internal:8086
       SERVICE_ITEM_MANAGER_URL: http://host.docker.internal:8081
       SPRING_REDIS_HOST: redis
-      ZUUL_DEBUG_REQUEST: 'true'
-      ZUUL_DEBUG_RESPONSE: 'true'
-      ZUUL_INCLUDE_DEBUG_HEADER: 'true'
```
Comment (bdrainer) [inline docker-compose.yml]: Moved these into the local configuration yml.

File: docker-compose.yml [hunk 2]
```diff
@@ -360,11 +357,10 @@ services:
...
     container_name: test-management-service
     image: fwsbac/tims-test-management-service:latest
-    healthcheck: *default-healthcheck
     ports:
-      - 8089:8080
-#    volumes:
-#      - ${HOME}:/home
+      - 8089:8089
+    volumes:
+      - ${HOME}:/home
```
Comment (bdrainer) [inline docker-compose.yml]: No sure why test-management changes are showing here but it not have rebased correctly.

File: item-authoring-tool/src/main/java/org/opentestsystem/ap/iat/config/SecurityConfig.java [hunk 3]
```diff
@@ -538,7 +536,7 @@ public FilterChainProxy samlFilter() throws Exception {
     }
 
     @Bean
-    public FilterRegistrationBean corsFilter() {
+    public CorsFilter corsFilter() {
```
Comment (bdrainer) [inline item-authoring-tool/src/main/java/org/opentestsystem/ap/iat/config/SecurityConfig.java]: There must have been a version change in Spring which is why this change was necessary.

File: item-management-service/src/main/resources/application-local.yml [hunk 4]
```diff
@@ -16,6 +16,13 @@ itembank:
   services:
     history:
       url: "${local.services.item-history-service.url:http://localhost:8086}"
+  aws:
+    endpointUrl: "${local.aws.endpoint.url:https://s3.us-east-2.amazonaws.com}"
+    region: "${local.aws.region:us-east-2}"
+    bucketName: "${local.aws.bucket.name}"
+    versionedBucketName: "${local.aws.bucket.name}-versioned"
+    accessKey: "${local.iat.aws.access.key}"
+    secretKey: "${local.iat.aws.access.secret}"
```
Comment (bdrainer) [inline item-management-service/src/main/resources/application-local.yml]: These were necessary for IMS.  Without it there was no way to override using the local values.

**Comment by bdrainer** [comment]
I am able to run things from IntelliJ, docker compse, and command line.  To run it from docker I chose to create a docker-tims.yml that lives next to the local-tims.yml

Each module has a root level application yml.  We should delete those at some point.   When we run things from the **command line** (i.e. not from IntelliJ or docker) we will need to set those env vars.  I created a file, _setenv_, that does that.  I run `source setenv` from the command line and it sets the 3 values below.  I didn't check this in but throw it out there as a suggestion. 

```
export SPRING_CONFIG_LOCATION=file://${HOME}/smarter_app/security/
export SPRING_CONFIG_NAME="application, local-tims"
export SPRING_PROFILES_ACTIVE=local
```

Below is my docker-tims.yml.  It only overrides what is needs to from the local-tims.yml . My .env file  has
`SPRING_CONFIG_NAME=application, local-tims, docker-tims` where the precedence is right to left with docker-tims having the highest precedence.

I left the env vars in the docker compose, but I'm not sure if those work.  I tested with the docker-tims and am not sure if I didn't have docker-tims if the env vars in docker-compose would be correct.  They look correct.  If we go the docker-tims route then we don't need to env vars in the docker-compose.

```
local:
  saml:
    keystore:
      file: "file:///home/smarter_app/iat-config-repo/ap-iat-ui.jks"
  services:
    eq-scoring-service:
      url: http://host.docker.internal:8184
    imrt-item-search-service:
      url: http://host.docker.internal:9080
    item-management-service:
      url: http://host.docker.internal:8081
    item-bulk-management-service:
      url: http://host.docker.internal:8088
    item-history-service:
      url: http://host.docker.internal:8086
    item-preivew-service:
      url: http://host.docker.internal:8200
    item-rendering-service:
      url: http://host.docker.internal:8083
    item-scoring-api:
      url: http://host.docker.internal:8077
    item-validation-service:
      url: http://host.docker.internal:8087
    permissions-service:
      url: http://host.docker.internal:8182
    tds-scoring-service:
      url: http://host.docker.internal:8183
    test-management-service:
      url: http://host.docker.internal:8089
    tims-scoring-service:
      url: http://host.docker.internal:8181
    wiris-engine:
      url: http://host.docker.internal:8300
```

Here is my current local-tims.yml.  I took out secrets.  Let me know if your's matches or if we should changes things to sync up.

```
local:
  aws:
    bucketName: iat-brett
    versionedBucketName: iat-brett-versioned
  db:
    iat:
      password: ""
    imrt:
      password: ""
  gitlab:
    access-token: 
    user: ap-system-user
    password: ""
    group: iat-brett
    webhook-url: ""
  iat:
    aws:
      access:
        key: ""
        secret: "" 
  rabbitmq:
    password: guest
  saml:
    keystore:
      file: "file:${HOME}/smarter_app/iat-config-repo/ap-iat-ui.jks"
      password: ""
    pke:
      alias: "sbtds.org"
      password: ""
    idp:
      url: "https://sso-deployment.sbtds.org/auth/saml2/jsp/exportmetadata.jsp?realm=/sbac"
    sp.entity.id: "ap-iat"
  security:
    basic:
      enabled: true
  services:
    eq-scoring-service:
      url: http://localhost:8184
    imrt-item-ingest:
      port: 9081
    imrt-item-search-service:
      port: 9080
      url: http://localhost:9080
    item-authoring-tool:
      port: 8080
    item-management-service:
      port: 8081
      url: http://localhost:8081
    item-bulk-management-service:
      port: 8088
      url: http://localhost:8088
    item-history-service:
      port: 8086
      url: http://localhost:8086
    item-preivew-service:
      port: 8200
      url: http://localhost:8200
    item-rendering-service:
      port: 8083
      url: http://localhost:8083
    item-scoring-api:
      port: 8077
      url: http://localhost:8077
    item-validation-service:
      port: 8087
      url: http://localhost:8087
    permissions-service:
      port: 8182
      url: http://localhost:8182
    tds-scoring-service:
      url: http://localhost:8183
    test-management-service:
      port: 8089
      url: http://localhost:8089
    tims-scoring-service:
      port: 8181
      url: http://localhost:8181
    wiris-engin:
      port: 8300
      url: http://localhost:8300
  tims:
    jwt.secret: 
```

File: item-authoring-tool/README.md [hunk 5]
```diff
@@ -1,192 +1,10 @@
...
-standing up the app dependencies.  
-
-The development profile routes are configured to match how docker compose stands 
-up the dependency.  
-
-Build `./gw build`
- 
-Run `./gw bootRun`
-
-# Frontend
```
Comment (malaffoon) [inline item-authoring-tool/README.md]: Is this frontend documentation somewhere else? It's not in the root documentation. It may make sense to leave this here since it is the only module that has front-end concerns.

File: item-authoring-tool/src/main/java/org/opentestsystem/ap/iat/config/SecurityConfig.java (hunk 3 above)
Comment (malaffoon) [reply item-authoring-tool/src/main/java/org/opentestsystem/ap/iat/config/SecurityConfig.java]: We did upgrade Spring from 1.5.x (for small x) to 1.5.21. I'm surprised this kind of change would be considered non-breaking.

File: item-authoring-tool/src/main/resources/application-local.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    item-viewer:
+      url: ${local.services.item-viewer.url:http://localhost:8079}
+    ims-api:
+      url: ${local.services.item-management-service.url:http://localhost:8081}/api
+    ibms-api:
+      url: ${local.services.item-bulk-management-service.url:http://localhost:8088}/api
+    irs-api:
+      url: ${local.services.item-rendering-service.url:http://localhost:8083}/api
+    ihs-api:
+      url: ${local.services.item-history-sercice.url:http://localhost:8086}/api
```
Comment (malaffoon) [inline item-authoring-tool/src/main/resources/application-local.yml]: typo, should be `item-history-service`

File: item-authoring-tool/src/main/resources/application-local.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    tims-scoring-api:
+      url: ${local.services.tims-scoring-service.url:http://localhost:8181}/api
+    auth-api:
+      url: ${local.services.permissions-service.url:http://localhost:8182}/api
+    admin-api:
+      url: ${local.services.permissions-service.url:http://localhost:8182}/api
+    preview-api:
+      url: ${local.services.item-preivew-service.url:http://localhost:8200}/iatpreview
+    wiris-engine:
+      url: ${local.services.wiris-engin.url:http://localhost:8300}/pluginwiris_engine
```
Comment (malaffoon) [inline item-authoring-tool/src/main/resources/application-local.yml]: typo i think

File: item-authoring-tool/src/main/resources/application.yml [hunk 8]
```diff
@@ -47,17 +47,10 @@ itembank:
 jwt:
   secret: "\${TIMS_JWT_SECRET}"
 
-logging:
-  level:
-    org.springframework.cloud.netflix: "trace"
```
Comment (malaffoon) [inline item-authoring-tool/src/main/resources/application.yml]: 👍

File: item-authoring-tool/src/main/resources/application.yml [hunk 9]
```diff
@@ -47,17 +47,10 @@ itembank:
...
-
 zuul:
   FormBodyWrapperFilter:
     pre:
       disable: true
-  debug:
-    request: true
-    response: true
```
Comment (malaffoon) [inline item-authoring-tool/src/main/resources/application.yml]: 👍   good cleanup

**Comment by malaffoon** [comment]
Yes, we need to provide instructions for running standalone (i.e. `gw bootRun` or `java -jar ...`); i have the beginnings in my local LOCAL-DEV.md file, so i'll flesh that out a bit and check it in. I'm hesitant to suggest environment variables since they will interfere with other projects, but i'll mention it since it will be the simplest approach.

Regarding docker: i'd like to make it so you don't need a special `docker-tims.yml` to make it work. That is, i'd like the docker-compose to have reasonable defaults for everything. Given your example, that may require a certain file layout for the jks file. But you really shouldn't need those services entries, they should be in the docker-compose file as needed. That said, we can and should document your approach of providing additional config names to customize an environment.

My local-tims.yml looks similar to yours except for two things: i haven't configured saml 'cause i haven't run the UI yet (no biggie), and i got rid of all the services entries (you shouldn't need them because the application-local.yml files have reasonable defaults.

**Comment by malaffoon** [comment]
Oh, and yes, the `application.yml` files in the module folders need to be removed at some point. I've been holding off until we've gleaned all the necessary info from them. Since most of them have config for pointing at `dev` and `uat` we should provide example config files for that. For example, `local-dev.yml` and `local-uat.yml` that have the datasource url and other settings in them.

File: item-authoring-tool/README.md (hunk 5 above)
Comment (bdrainer) [reply item-authoring-tool/README.md]: I'll add the frontend stuff back.
